### PR TITLE
Automatic update for keyboard atlas references

### PR DESCRIPTION
**Pull request by Zod-**
Overview
---
Updated the keyboard atlas to reference the changed sprites.
Also adds an editor script that will automatically keep the atlas up to date by getting all sprites used by the keyboard prefab and updating the atlas with them.
The atlas will be updated when:
* The editor gets started
* The prefab gets updated
* A sprite on the atlas gets deleted
* A atlas reference gets deleted/updated

Changes
---
- Fixes: #1665

### CODE REVIEW

File: Assets/HoloToolkit/Utilities/Scripts/AtlasReferenceUpdater/Editor/Extensions/SpriteAtlasExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.U2D;
+
+namespace HoloToolkit.Unity
+{
+    public static class SpriteAtlasExtensions
+    {
+        public static string SpritePackables = "m_EditorData.packables";
```
Comment (StephenHodgson) [inline Assets/HoloToolkit/Utilities/Scripts/AtlasReferenceUpdater/Editor/Extensions/SpriteAtlasExtensions.cs]: This value is from Unity's serialization?
Should we make this const?

File: Assets/HoloToolkit/Utilities/Scripts/AtlasReferenceUpdater/Editor/AtlasReferenceUpdater.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            if (atlasReference != null)
+            {
+                EditorApplication.delayCall += FindAtlasPrefabReferences;
+                return AssetDeleteResult.DidNotDelete;
+            }
+
+            var deletedSprite = AssetDatabase.LoadAssetAtPath<Sprite>(assetPath);
+            if (deletedSprite != null)
+            {
+                foreach (var reference in References.Where(re => re.Atlas.ContainsSprite(deletedSprite)))
```
Comment (StephenHodgson) [inline Assets/HoloToolkit/Utilities/Scripts/AtlasReferenceUpdater/Editor/AtlasReferenceUpdater.cs]: Is re a reference?

Comment (Zod-) [reply Assets/HoloToolkit/Utilities/Scripts/AtlasReferenceUpdater/Editor/AtlasReferenceUpdater.cs]: yes

File: Assets/HoloToolkit/Utilities/Scripts/AtlasReferenceUpdater/Editor/Extensions/SpriteAtlasExtensions.cs (hunk 1 above)
Comment (Zod-) [reply Assets/HoloToolkit/Utilities/Scripts/AtlasReferenceUpdater/Editor/Extensions/SpriteAtlasExtensions.cs]: Oops yes, it was supposed to be const.

File: Assets/HoloToolkit/Utilities/Scripts/AtlasReferenceUpdater/Editor/AtlasReferenceUpdater.cs (hunk 2 above)
Comment (StephenHodgson) [reply Assets/HoloToolkit/Utilities/Scripts/AtlasReferenceUpdater/Editor/AtlasReferenceUpdater.cs]: I think it's okay. But something like `_ref` or `_reference` is okay.  Up to you.

**Pull request merged**